sqlparser/lib/src/analysis/steps/nested_query_resolver.dart

```diff
@@ -0,0 +1,68 @@
+part of '../analysis.dart';
+
+/// Converts all references in nested queries, that require data from the
+/// parent query into variables.
+class NestedQueryResolver extends RecursiveVisitor<void, void> {
+  final AnalysisContext context;
+
+  NestedQueryResolver(this.context);
+
+  @override
+  void visitMoorSpecificNode(MoorSpecificNode e, void arg) {
+    if (e is NestedQueryColumn) {
+      _transform(context, e);
+    } else {
+      super.visitMoorSpecificNode(e, arg);
+    }
+  }
+}
+
+void _transform(AnalysisContext context, NestedQueryColumn e) {
+  e.select.transformChildren(_NestedQueryTransformer(context), null);
+
+  AstPreparingVisitor.resolveIndexOfVariables(
+    e.allDescendants.whereType<Variable>().toList(),
+  );
+}
+
+class _NestedQueryTransformer extends Transformer<void> {
+  final AnalysisContext context;
+
+  _NestedQueryTransformer(this.context);
+
+  @override
+  AstNode? visitReference(Reference e, void arg) {
+    // if the scope of the nested query cannot resolve the reference, the
+    // reference needs to be retrieved from the parent query
+    if (e.entityName != null &&
+        e.scope.resolve(e.entityName!, includeParents: false) == null) {
+      final result = e.scope.resolve(e.entityName!);
```

Comment:
Instead of checking scopes manually, have you seen `Reference.resultEntity`? You can use it to get access to the AST node responsible for adding something to a scope, which might be a bit easier to check?

I haven't tried it, but does something like `e.resultEntity.origin.scope != e.scope` work (modulo null handling for unresolved references)?

Reply:
I tried it and I think your method also works. Should I remove the `includeParents` flag then?

Reply:
Okey nvm it is not working, somehow resultEntity is always null and I can’t find where it should be set to some value?

Reply:
If I set resultEntity in the `ReferneceResolver` like this: 
```dart
void _resolveReferenceInTable(Reference ref, ResultSet resultSet,
    {ResultSetAvailableInStatement? source}) {
  var column = resultSet.findColumn(ref.columnName);
  if (column == null) {
    _reportUnknownColumnError(ref, columns: resultSet.resolvedColumns);
  } else {
    if (source != null) {
      column = AvailableColumn(column, source);
    }
    ref.resolved = column;
    ref.resultEntity = source;
  }
}
```
Then it works :)

Reply:
Oh yeah, it should absolutely be set there. Good catch!